lib/src/bootloader.rs

```diff
@@ -26,3 +26,94 @@ pub(crate) fn install_via_bootupd(
         .verbose()
         .run()
 }
+
+#[context("Installing bootloader using zipl")]
+pub(crate) fn install_via_zipl(device: &Utf8Path, boot_uuid: &str) -> Result<()> {
+    // Identify the target boot partition from UUID
+    let fs = crate::mount::inspect_filesystem_by_uuid(boot_uuid)?;
+    let boot_dir = Utf8Path::new(&fs.target);
+    let maj_min = fs.maj_min;
+
+    // Ensure that the found partition is a part of the target device
+    crate::blockdev::list_dev(device)?
+        .children
+        .with_context(|| format!("no partition found on {device:?}"))?
+        .iter()
+        .find(|dev| dev.maj_min == Some(maj_min.clone()))
```

Comment:
Minor nit, we're cloning the string for each iteration here. I think you may have done this because of a mismatch between `&String` and `String`? The best fix is to coerce both to `&str`:

`.find(|dev| dev.maj_min.as_deref() == Some(maj_min.as_str())`
I think will work.

Reply:
done